assemblyline/common/entropy.py

```diff
@@ -24,13 +28,26 @@ def calculate_partition_entropy(fin: BinaryIO, num_partitions: int = 50) -> Tupl
     fin.seek(0, io.SEEK_END)
     size = fin.tell()
     fin.seek(0)
-    partition_size = int(ceil(size / float(num_partitions)))
+
+    if size == 0:
+        return 0, [0]
+
+    # Calculate the partition size to get the desired amount of partitions but make sure those
+    # partitions are the minimum partition size
+    partition_size = max((size - 1)//num_partitions + 1, MIN_PARTITION_SIZE)
+
+    # If our calculated partition size is the minimum partition size, our files is likely too small we will
+    # calculate an alternate partition size that will make sure all blocks are equal size
+    if partition_size == MIN_PARTITION_SIZE:
+        partition_size = (size-1) // ((size-1)//partition_size + 1) + 1
```

Comment:
Since the value is immediately reassigned if `max` picks `MIN_PARTITION_SIZE` we can simplify by using:
```
if partition_size <= MIN_PARTITION_SIZE:
    partition_size = (size -1) // ((size - 1) // MIN_PARTITION_SIZE + 1) + 1
```